spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitAdminTests.java

```diff
@@ -194,4 +204,96 @@ private void cleanQueuesAndExchanges(RabbitAdmin rabbitAdmin) {
 		rabbitAdmin.deleteExchange("testex.all");
 	}
 
+	@Test
+	public void testMultiEntities() {
+		ConfigurableApplicationContext ctx = new AnnotationConfigApplicationContext(Config.class);
+		RabbitTemplate template = ctx.getBean(RabbitTemplate.class);
+		template.convertAndSend("e1", "k1", "foo");
+		template.convertAndSend("e2", "k2", "bar");
+		template.convertAndSend("e3", "k3", "baz");
+		template.convertAndSend("e4", "k4", "qux");
+		assertEquals("foo", template.receiveAndConvert("q1"));
+		assertEquals("bar", template.receiveAndConvert("q2"));
+		assertEquals("baz", template.receiveAndConvert("q3"));
+		assertEquals("qux", template.receiveAndConvert("q4"));
+		RabbitAdmin admin = ctx.getBean(RabbitAdmin.class);
+		admin.deleteQueue("q1");
+		admin.deleteQueue("q2");
+		admin.deleteQueue("q3");
+		admin.deleteQueue("q4");
+		admin.deleteExchange("e1");
+		admin.deleteExchange("e2");
+		admin.deleteExchange("e3");
+		admin.deleteExchange("e4");
+		ctx.close();
+	}
+
+	@Configuration
+	public static class Config {
+
+		@Bean
+		public ConnectionFactory cf() {
+			return new CachingConnectionFactory("localhost");
+		}
+
+		@Bean
+		public RabbitAdmin admin(ConnectionFactory cf) {
+			return new RabbitAdmin(cf);
+		}
+
+		@Bean
+		public RabbitTemplate template(ConnectionFactory cf) {
+			return new RabbitTemplate(cf);
+		}
+
+		@Bean
+		public DirectExchange e1() {
+			return new DirectExchange("e1", false, true);
+		}
+
+		@Bean
+		public Queue q1() {
+			return new Queue("q1", false, false, true);
+		}
+
+		@Bean
+		public Binding b1() {
+			return BindingBuilder.bind(q1()).to(e1()).with("k1");
+		}
+
+		@Bean
+		public List<Exchange> es() {
+			return Arrays.asList(new Exchange[] {
+					new DirectExchange("e2", false, true),
+					new DirectExchange("e3", false, true)
+			});
+		}
+
+		@Bean
+		public List<Queue> qs() {
+			return Arrays.asList(new Queue[] {
```

Comment:
Overdid from the `Arrays.asList` perspective. It is vararg method :smile: .
Fixing during merge